active/000-buisness-attributes/000-business-attributes.md

```diff
@@ -0,0 +1,258 @@
+* Allow dataset owners to partially inherit from business attributes (eg only inherit description but not tags)
+* Allow fields to have multiple business attributes
+* Update descriptions in data sources
+
+## Detailed design
+
+### Sample Mockups
+
+#### Show Business Atrribute Launcher 
+![](business-attribute-rfc-1.png)
+
+#### List Business Atrributes
+![](business-attribute-rfc-2.png)
+
+#### Create Business Attribute  
+![](business-attribute-rfc-3.png)
+
+#### Business Attribute Details
+![](business-attribute-rfc-4.png)
+
+#### Enrich Schema
+Current feature of enriching the schema field of dataset remains as it is
+
+![](business-attribute-rfc-5.png)
+
+#### Attaching Business Attribute
+![](business-attribute-rfc-6.png)
+
+#### Metadata Model Enhancements
+
+Below are the current Relationships of Dataset Entity with its aspects in Datahub.
+
+![](business-attribute-rfc-7.png)
+
+
+### Model Business Attribute Entity
+
+We are proposing to introduce a new entity `BusinessAttribute`, which will use a new Aspect `BusinessAttributeInfo`. This new entity will be a top level entity to allow it to be defined and controlled independently by the business team.
+
+User can able to attach `Business Attributes` only to `Schema Fields`. We will make necessary UI changes to control this feature.
+
+New `BusinessAttributeInfo` aspect contains existing `EditableSchemaFieldInfo` record type and also `customProperties`.  Also we are introducing one new record type, `PhysicalEditableSchemaFieldInfo` which includes existing `EditableSchemaFieldInfo`. As per this implementation, existing aspect `EditableSchemaMetadata` now contains a list of `PhysicalEditableSchemaFieldInfo` records instead of `EditableSchemaFieldInfo` records.
+
+Each `PhysicalEditableSchemaFieldInfo` will contain an `EditableSchemaFieldInfo` and the new field "`businessAttribute`" to represent the link between dataset schema field and business attribute. We are also proposing the introduction of a new field, "`type`", in `EditableSchemaFieldInfo`.
+
+### URN Representation for Business Attribute
+![](business-attribute-rfc-8.png)
+
+```
+urn:li:businessAttribute:b3916dfe-a27c-4916-97be-60773dca90d7
+```
+
+###  Business Attribute Attachment to SchemaField
+When the user attaches a business attribute to the corresponding schema field, the link will be represented by the business attribute id in the `PhysicalEditableSchemaFieldInfo` for Dataset. 
+
+Any changes made to business attribute will be reflected on dataset details page on the fly. Also, delinking the business attribute from the schema field is implemented by simply removing the business attribute reference.
+
+### Metadata Graph
+
+#### Enabling Capability of searching Dataset entities as per Description/tags of Business Attributes
+
+We are proposing to introduce new annotation "`@SearchableRef`", through which we can populate the elastic indexes with expanded details regarding the referenced entity.
```

Comment:
The write fanout in this case would be the main issue with this approach. Have you investigated whether elastic can offer any kind of join-like capabilities for us to maintain separate documents for the dataset entity index and the business attribute entity index while still being able to drive searches to match on descriptions etc from business attributes to return dataset docs?

Reply:
Our investigation reveals that due to its distributed nature, join operations are generally not advised and can be costly in Elasticsearch, which performs optimally with denormalized data. As a result, we suggest incorporating business attribute properties into the datasets index using PlatformEvents. This approach ensures eventual consistency in search results, while changes are instantly reflected on the user interface.